### PR TITLE
Use child type in some places to make it more clear

### PR DESCRIPTION
**Pull request by revans2**
We used "nested" types in a number of places to refer to a "child" type. This was confusing and this tries to fix it. If needed I can add in a definition of a nested type and a child type to the docs too.

### CODE REVIEW

**Comment by revans2** [comment]
build

**Comment by tgravescs** [comment]
child makes it more clear to me but should get others feedback and keep things consistent in our terminology.  @sameerz

**Review by jlowe — DISMISSED** [review]

File: sql-plugin/src/main/scala/com/nvidia/spark/rapids/TypeChecks.scala [hunk 1]
```diff
@@ -144,18 +144,18 @@ final class TypeSig private(
    */
   def withAllLit(): TypeSig = {
     // don't need to combine initialTypes with litOnlyTypes because litOnly should be a subset
-    new TypeSig(initialTypes, maxAllowedDecimalPrecision, nestedTypes, initialTypes, notes)
+    new TypeSig(initialTypes, maxAllowedDecimalPrecision, childTypes, initialTypes, notes)
   }
 
   /**
-   * Combine two type signatures together.  Base types and nested types will be the union of
+   * Combine two type signatures together.  Base types and child types will be the union of
```
Comment (ttnghia) [inline sql-plugin/src/main/scala/com/nvidia/spark/rapids/TypeChecks.scala:151]:
```suggestion
   * Combine two type signatures together. Base types and child types will be the union of
```

**Comment by ttnghia** [comment]
> child makes it more clear to me but should get others feedback and keep things consistent in our terminology. @sameerz

Does `childType` here really mean a subtype or a type of some children (column)?

**Comment by revans2** [comment]
> Does `childType` here really mean a subtype or a type of some children (columns) or something similar?

Type of child columns. Not child type as in the `IntegerType` class is a child of the `NumericType` class, which is also a child of the `AtomicType` class in Spark.

**Comment by revans2** [comment]
build

**Review by jlowe — DISMISSED** [review]

**Comment by revans2** [comment]
build

**Review by jlowe — DISMISSED** [review]

**Comment by revans2** [comment]
build

**Comment by revans2** [comment]
build

**Review by jlowe — APPROVED** [review]

**Comment by sameerz** [comment]
build

**Comment by pxLi** [comment]
build

**Pull request merged**